src/Commands/Aks/BaseAksCommand.cs

```diff
@@ -0,0 +1,59 @@
+using System.Diagnostics.CodeAnalysis;
+using AzureMcp.Commands.Subscription;
+using AzureMcp.Models.Option;
+using AzureMcp.Options.Aks;
+
+namespace AzureMcp.Commands.Aks;
+
+public abstract class BaseAksCommand<
+    [DynamicallyAccessedMembers(TrimAnnotations.CommandAnnotations)] TOptions>
+    : SubscriptionCommand<TOptions> where TOptions : BaseAksOptions, new()
+{
+    protected new readonly Option<string> _resourceGroupOption = OptionDefinitions.Common.ResourceGroup;
+    protected readonly Option<string> _clusterOption = OptionDefinitions.Aks.Cluster;
+
+    // Allow derived classes to specify if resource group is required
+    protected virtual bool RequiresResourceGroup => true;
+
+    protected override void RegisterOptions(Command command)
+    {
+        base.RegisterOptions(command);
+
+        // Create a copy of the resource group option with the required setting from derived class
+        var resourceGroupOption = new Option<string>(
+            _resourceGroupOption.Name,
+            _resourceGroupOption.Description!)
+        {
+            IsRequired = RequiresResourceGroup
+        };
+        command.AddOption(resourceGroupOption);
```

Comment:
Can you explain this a bit more?  You should be able to use the resource group option from Global Options and then update if it is required directly on that option.